embedded-graphics/src/primitives/circle.rs

```diff
@@ -132,28 +134,23 @@ where
 {
     type Item = Pixel<C>;
 
-    // https://stackoverflow.com/questions/1201200/fast-algorithm-for-drawing-filled-circles
+    // https://stackoverflow.com/a/1237519/383609
+    // https://stackoverflow.com/questions/1201200/fast-algorithm-for-drawing-filled-circles#comment80182898_1237519
     fn next(&mut self) -> Option<Self::Item> {
         // If border or stroke colour is `None`, treat entire object as transparent and exit early
         if self.style.stroke_color.is_none() && self.style.fill_color.is_none() {
             return None;
         }
 
-        let radius = self.radius as i32 - self.style.stroke_width_i32() + 1;
-        let outer_radius = self.radius as i32;
-
-        let radius_sq = radius * radius;
-        let outer_radius_sq = outer_radius * outer_radius;
-
         loop {
             let t = self.p;
-            let len = t.x * t.x + t.y * t.y;
+            let len = t.x.pow(2) + t.y.pow(2);
 
-            let is_border = len > radius_sq - radius && len < outer_radius_sq + radius;
+            let is_stroke = len > self.inner_threshold && len <= self.outer_threshold;
 
-            let is_fill = len <= outer_radius_sq + 1;
+            let is_fill = len <= self.outer_threshold;
```

Comment:
I just tried this but the `tiny_circle_filled` test failed with:

```
###
###
##
```

I think the original is ok because the `if` below checks for stroke before fill, so no overdraw occurs AFAIK.